landoapi/auth.py

```diff
@@ -357,11 +371,19 @@ class require_auth0:
     accessed using flask.g.auth0_user.
     """
 
-    def __init__(self, scopes: Optional[Iterable[str]] = None, userinfo: bool = False):
+    def __init__(
+        self,
+        groups: Optional[Iterable[str]] = None,
+        scopes: Optional[Iterable[str]] = None,
+        userinfo: bool = False,
+    ):
         assert scopes is not None, (
             "`scopes` must be provided. If this endpoint truly does not "
             "require any scopes, explicilty pass an empty tuple `()`"
         )
+        assert not groups or userinfo, "Requiring `groups` implies `userinfo`."
```

Comment:
Shouldn't use asserts in code flow. Instead, these should be changed to if statements / exceptions raised.

Reply:
I copied this pattern from the `assert` statement that already exists for `scopes`. I'm happy to change it as I usually don't use `assert` in code flows either. However I'll note that in this case, the code flow that is caught is a programming error instead of something that would be hit due to runtime conditions. So this code path would always throw in the tests when someone on our team uses the decorator incorrectly. That might explain why an `assert` is used here instead of an `if` statement.

Reply:
I think this would be better refactored to handle required arguments a little more conventionally. The above code (re: `scopes`) also does not make much sense. If an argument is required, it should not be marked as `Optional` and should be changed to a positional argument.

Reply:
The `scopes` assert is fixed in #311, and I'll make the check here an `if` statement instead.